logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/log/LogStorageAppender.java

```diff
@@ -178,10 +186,19 @@ public void onActorFailed() {
   }
 
   private void onWriteBufferAvailable() {
-    final BlockPeek blockPeek = new BlockPeek();
-    if (writeBufferSubscription.peekBlock(blockPeek, maxAppendBlockSize, true) > 0) {
-      appendBlock(blockPeek);
-    } else {
+    final var blockPeek = new BlockPeek();
+    final var readBytes = writeBufferSubscription.peekBlock(blockPeek, maxAppendBlockSize, true);
+
+    final var shouldAppendBlock = readBytes > 0;
+    var appendBlockSucceeded = false;
+
+    if (shouldAppendBlock) {
+      appendBlockSucceeded = appendBlock(blockPeek);
+    }
+
+    // Yield the thread, when no bytes are read or
+    // when the block could not be appended.
+    if (!shouldAppendBlock || !appendBlockSucceeded) {
```

Comment:
:question: Is it necessary to yield if we not append? I think the subscription (actor condition) is only called if it was notified right?

Reply:
Yes, if there is nothing to append, it is still necessary to yield, because of

https://github.com/camunda-cloud/zeebe/blob/3da1a75d52e524b123f91a8ec9e480a0337bbf42/util/src/main/java/io/camunda/zeebe/util/sched/channel/ChannelConsumerCondition.java#L43-L48

In more detail, because of that condition `polledCount > processedTiggersCount` (BTW there is a typo in the source code 😄 ). There might be a case, where the `polledCount` is much greater than the `processedTiggersCount` but currently there is nothing to peek from the write buffer. Meaning, the condition in `#poll()` still returns `true`. If the yield would not happen, then the task would continue executing the corresponding actor job until the condition is `false` (i.e., the `processedTiggersCount` is greater or equal to `polledCount`).